containerd-shim-spin/src/engine.rs

```diff
@@ -533,10 +552,60 @@ fn trigger_command_for_resolved_app_source(resolved: &ResolvedAppSource) -> Resu
     Ok(trigger_types.iter().map(|x| x.to_string()).collect())
 }
 
+// For each Spin app variable, checks if a container environment variable with
+// the same name exists and duplicates it in the environment with the
+// application variable prefix
+fn configure_application_variables_from_environment_variables(
+    resolved: &ResolvedAppSource,
+) -> Result<()> {
+    let variables: Vec<_> = resolved
+        .variables()
+        .into_iter()
+        .map(|v| v.to_ascii_uppercase())
+        .collect();
+    for variable in variables {
+        let prefixed = format!("{}_{}", SPIN_APPLICATION_VARIABLE_PREFIX, variable);
+        if let Ok(val) = env::var(variable) {
+            env::set_var(prefixed, val);
+        }
+    }
+    Ok(())
+}
+
 #[cfg(test)]
 mod tests {
+    use std::env;
+
     use super::*;
 
+    #[test]
+    fn test_configure_application_variables_from_environment_variables() {
+        let app_json = r#"
+        {
+            "spin_lock_version": 1,
+            "entrypoint": "test",
+            "components": [],
+            "variables": {"should_be_prefixed": { "required": "true"},  "do_not_reset" : { "required": "true"}, "not_set_as_container_env": { "required": "true"}, "ignored_if_not_uppercased_env": { "required": "true"}},
+            "triggers": []
+        }"#;
+        let locked_app = LockedApp::from_json(app_json.as_bytes()).unwrap();
+        let resolved = ResolvedAppSource::OciRegistry { locked_app };
+        env::set_var("SPIN_VARIABLE_DO_NOT_RESET", "val1");
```

Comment:
**suggestion:** use [temp-env](https://crates.io/crates/temp-env) to set environment variables temporarily in tests.

Reply:
Great suggestion. Thank you!